### PR TITLE
[Kotlin Migration] PreferenceExtensionsTest

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Yet another test of https://github.com/ankidroid/Anki-Android/pull/10779

### CODE REVIEW

File: AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt [hunk 1]
```diff
@@ -13,91 +13,80 @@
...
+import org.mockito.Mockito
+import org.mockito.MockitoAnnotations
+import java.lang.RuntimeException
+import java.util.function.Supplier
+
+// Unknown issue: @CheckResult should provide warnings on this class when return value is unused, but doesn't.
+class PreferenceExtensionsTest {
     @Mock
-    private SharedPreferences mMockPreferences;
+    private val mMockPreferences: SharedPreferences? = null
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt:34]: cleanup: see if we can `lateinit` the `@Mock`s

File: AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt [hunk 2]
```diff
@@ -13,91 +13,80 @@
...
     @Before
-    public void setUp() {
-        MockitoAnnotations.openMocks(this);
-        Mockito.when(mMockPreferences.contains(VALID_KEY)).thenReturn(true);
-        Mockito.when(mMockPreferences.getString(eq(VALID_KEY), anyString())).thenReturn(VALID_RESULT);
-        Mockito.when(mMockPreferences.edit()).thenReturn(mMockEditor);
-        Mockito.when(mMockEditor.putString(anyString(), anyString())).thenReturn(mMockEditor);
+    fun setUp() {
+        MockitoAnnotations.openMocks(this)
+        Mockito.`when`(mMockPreferences!!.contains(VALID_KEY)).thenReturn(true)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt:45]: cleanup: `whenever`

File: AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt [hunk 3]
```diff
@@ -13,91 +13,80 @@
...
+        Mockito.`when`(mMockPreferences.getString(ArgumentMatchers.eq(VALID_KEY), ArgumentMatchers.anyString())).thenReturn(VALID_RESULT)
+        Mockito.`when`(mMockPreferences.edit()).thenReturn(mMockEditor)
+        Mockito.`when`(mMockEditor!!.putString(ArgumentMatchers.anyString(), ArgumentMatchers.anyString())).thenReturn(mMockEditor)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt:48]: fix: import `ArgumentMatchers`

File: AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt [hunk 4]
```diff
@@ -13,91 +13,80 @@
...
+    private val forMissingKey: String?
+        get() = getOrSetString(MISSING_KEY) { LAMBDA_RETURN }
 
     @Test
-    public void existingKeyReturnsMappedValue() {
-        String ret = getOrSetString(VALID_KEY, UNUSED_SUPPLIER);
-        assertEquals(ret, VALID_RESULT);
+    fun existingKeyReturnsMappedValue() {
+        val ret = getOrSetString(VALID_KEY, UNUSED_SUPPLIER)
+        Assert.assertEquals(ret, VALID_RESULT)
```
Comment (david-allison) [inline AnkiDroid/src/test/java/com/ichi2/preferences/PreferenceExtensionsTest.kt:57]: fix: import `Assert`

**Review by david-allison — CHANGES_REQUESTED** [review]
Some fixes, some cleanups

commit messages still need work

**Comment by lukstbit** [comment]
@mikehardy Can be closed because it was already done in #11241

**Pull request closed without merging**